dbt/profiles.yml

```diff
@@ -3,8 +3,9 @@ athena:
   outputs:
     dev:
       type: athena
-      s3_staging_dir: s3://ccao-dbt-athena-test-us-east-1/results/
-      s3_data_dir: s3://ccao-dbt-athena-test-us-east-1/data/
+      s3_staging_dir: s3://ccao-dbt-athena-dev-us-east-1/results/
+      s3_data_dir: s3://ccao-dbt-athena-dev-us-east-1/data/
+      s3_data_naming: schema_table
```

Comment:
See [the docs](https://dbt-athena.github.io/docs/configuration/table-configuration#table-data-location) for a detailed explanation of the S3 data naming options that `dbt-athena` exposes to us. `schema_table` corresponds to `{s3_data_dir}/{schema}/{table}/`, which matches how production data has historically been stored, so I figured we should move to it over the default `schema_table_unique`. This config renders the `external_location` attribute of the CTAS unnecessary.

Reply:
Nice! This is a great simplification. Big fan.

**question (blocking):** What is the materialization strategy used here? As in, if someone is working on a branch and creates a new materialized table, then edits their branch and _re-_ materializes it, will the old table be entirely overwritten? Will it delete all the partitions created by the previous materialization?

Reply:
Under `schema_table`, new materialized tables will indeed write over the old ones! My thinking was that this was desirable in order to save us on S3 space, but we could also use `schema_table_unique` in development if we want tables to be persisted between versions.

Reply:
@jeancochrane Just to call out some potentially unexpected behavior, some quick testing shows that dbt doesn't necessarily remove the entire old materialization:

Here we create a new test table partitioned by year and stored in S3 as parquet:

```sql
CREATE TABLE location.test_table
WITH (
    format = 'Parquet',
    write_compression = 'SNAPPY',
    external_location = 's3://ccao-dbt-athena-test-us-east-1/test_table',
    partitioned_by = ARRAY['year']
) AS
SELECT b23025_007m, year
FROM census.acs5
```

Dropping the table removes the table _metadata_ defining the location and partitions, but _not_ the data itself. All the data still exists in S3.

```sql
DROP TABLE location.test_table;
```

Re-instantiating the table using a subset of the original partitions correctly creates _new_ parquet files for each year after 2017.

```sql
CREATE TABLE location.test_table
WITH (
    format = 'Parquet',
    write_compression = 'SNAPPY',
    external_location = 's3://ccao-dbt-athena-test-us-east-1/test_table',
    partitioned_by = ARRAY['year']
) AS
SELECT b23025_007m, year
FROM census.acs5
WHERE year >= '2018'

SELECT year, COUNT(*) AS cnt
FROM location.test_table
GROUP BY year
ORDER BY year
```

row | year | cnt
-- | -- | --
1 | 2018 | 17521
2 | 2019 | 17521
3 | 2020 | 17619
4 | 2021 | 17619

HOWEVER, the original pre-2018 partitions/parquet files _remain in S3_, since they aren't overwritten by the new `CREATE TABLE` nor actually dropped by the `DROP TABLE`. This doesn't matter too much since those partitions aren't included in the Athena table definition, but we do use a lot of crawlers (which would definitely add those stale partitions back to the table).

I'm not sure this is actually a problem or that there's anything to actually do about it. I just felt the need to call it out since it makes me uneasy and is not how a typical relational DB would act.

Reply:
Hm, interesting! Thanks for the experiment @dfsnow, I'm going to play around a little bit to see if any of the other materialization options perform more closely to expectation here. I'm guessing you ran the `DROP TABLE` statement directly in Athena; did you build `location.test_table` using dbt or using an Athena query? If the former, do you happen to have copies of the dbt models that generated those parquet files?

Reply:
I didn't build directly with dbt no, but happy to test further! Though I'd be really surprised if `dbt-athena` had some sort of logic to drop the extra partitions. Either way, let's not consider this blocking, just something to keep in mind.